src/components/Menu/Menu.stories.tsx

```diff
@@ -20,90 +20,129 @@ export default {
-          Not Possible (disabled)
-        </Menu.Item>
-      </Menu.Items>
-    </Menu>
-  ),
+  args: {
+    children: (
+      <>
+        <Menu.Button>
+          Long Trigger Button Text to Demonstrate Popover Matching
+        </Menu.Button>
+        <Menu.Items data-testid="menu-content">
+          <Menu.Item
+            href="https://headlessui.com/react/menu#menu-button"
+            icon="link"
+          >
+            Headless UI Docs
+          </Menu.Item>
+          <Menu.Item
+            href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu"
+            icon="link"
+          >
+            MDN: Menu
+          </Menu.Item>
+          {/* eslint-disable-next-line no-alert */}
+          <Menu.Item onClick={() => alert('Item clicked')}>
+            Trigger Action
+          </Menu.Item>
+          <Menu.Item disabled href="https://example.org/" icon="warning">
+            Not Possible (disabled)
+          </Menu.Item>
+        </Menu.Items>
+      </>
+    ),
+  },
 };
 
 export const WithShortButtonText: StoryObj<MenuProps> = {
-  render: (args) => (
-    <Menu {...args}>
-      <Menu.Button>Menu</Menu.Button>
-      <Menu.Items data-testid="menu-content">
-        <Menu.Item
-          href="https://headlessui.com/react/menu#menu-button"
-          icon="link"
-        >
-          Headless UI Docs
-        </Menu.Item>
-        <Menu.Item
-          href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu"
-          icon="link"
-        >
-          MDN: Menu
-        </Menu.Item>
-        {/* eslint-disable-next-line no-alert */}
-        <Menu.Item onClick={() => alert('Item clicked')}>
-          Trigger Action
-        </Menu.Item>
-        <Menu.Item disabled href="https://example.org/" icon="warning">
-          Not Possible (disabled)
-        </Menu.Item>
-      </Menu.Items>
-    </Menu>
-  ),
+  args: {
+    children: (
+      <>
+        <Menu.Button>Menu</Menu.Button>
+        <Menu.Items data-testid="menu-content">
+          <Menu.Item
+            href="https://headlessui.com/react/menu#menu-button"
+            icon="link"
+          >
+            Headless UI Docs
+          </Menu.Item>
+          <Menu.Item
+            href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu"
+            icon="link"
+          >
+            MDN: Menu
+          </Menu.Item>
+          {/* eslint-disable-next-line no-alert */}
+          <Menu.Item onClick={() => alert('Item clicked')}>
+            Trigger Action
+          </Menu.Item>
+          <Menu.Item disabled href="https://example.org/" icon="warning">
+            Not Possible (disabled)
+          </Menu.Item>
+        </Menu.Items>
+      </>
+    ),
+  },
+};
+
+export const WithCustomButton: StoryObj<MenuProps> = {
+  args: {
+    children: (
+      <>
+        <Menu.Button showExpandIcon={false} status="neutral" variant="icon">
+          <div className="fpo !py-0">Menu Button</div>
+        </Menu.Button>
```

Comment:
I do believe there is an abstraction that can further simplify this, but I havent had the time to think thru it properly. As a hint tho, since we have control over what `Menu.Button` is, we could also specify additional `.Button` subcomponents that apply the desired props when used, so that consumers don't have to remember these prop values, class overrides, etc.

For example, something like:

```jsx
<Menu.PlainButton>
    <div className="fpo">... or an icon, or an avatar component, etc.</div>
</Menu.PlainButton>
```

In this case, whatever is in `PlainButton` (coming from a properly accessible design) takes over all the duties, and this essentially exposes the untreated HeadlessUI button.

This wouldn't block the current PR. If such an approach makes sense, it is additive and would mean we could later deprecate anything it would replace.

Reply:
ooh interesting. I also just realized that the Button variants (especially `"icon"`) don't work properly because the menu's built-in button styles are overriding it. Ideally we want to make this "plain" styling case really easy -- Andrew's suggestion would work, or something like `<Menu.Button variant="plain">`.

Using the EDS Button underneath still makes sense, but I'm now questioning if we should allow consumers to pass through all Button props ... may be better/less confusing for us to control the few cases at the Menu.Button level.

Reply:
I'm liking the PlainButton api which just ships the Headless Menu Button. Not sure about the button variants as well. We can just go back to not accepting button props and if customization is wanted, consumers can use the PlainButton?

Reply:
One other benefit of having a `"plain"` variant on `Button`/`ClickableStyle` is that it can manage implementing the focus ring. Ex:

```jsx
<Button variant="plain">
   <Avatar user={...} />
</Button>
```

would automatically allow a properly-spaced focus ring clamped around the shape of the `children` (or could have Tailwind utilities provided to change the wrapper shape to get a circle). `PlainButton` could allow `as` so the consumer can implement the entire button if we ship the headless wrapper. I'm not sure I follow the button prop bit, but we can chat thru that if it still is relevant

Reply:
Think you commented this just as I pushed a commit with `PlainButton` haha. I do like the `as` prop.
The button prop was allowing MenuButton to accept Button props for further styling, but I removed that since that's possible with PlainButton now.